src/game/Spell.cpp

```diff
@@ -4193,6 +4231,9 @@ SpellCastResult Spell::CheckCast(bool strict)
             return SPELL_FAILED_NOT_READY;
     }
 
+    if (!m_caster->isAlive() && m_caster->GetTypeId() == TYPEID_PLAYER && !m_spellInfo->HasAttribute(SPELL_ATTR_CASTABLE_WHILE_DEAD))
```

Comment:
I dont like doing these, because the benefit is minimal, and its HORRIBLE for readability